decoder.go

```diff
@@ -58,6 +61,26 @@ func (vc videoCodec) String() string {
 	}
 }
 
+type avFrameWrapper struct {
```

Comment:
1. How many `avFrameWrapper` do we expect to have for a given model instance?
2. When do we expect https://github.com/erh/viamrtsp/pull/41/files#diff-2a967f84f1af0d52111b8dbfbeadc44bfeb1092dacebb2bc0648a0e57284c04bR76 to run? AKA, when do we expect the `avFrameWrapper` to be garbage collected? How frequently does that happen?
3. Why is `allocateAVFrame` called both within pool.New and newDecoder? 
4. It seems like not all `allocateAVFrame` are managed by the pool, such as the one created in newDecoder. Why is that? How do I know if an avFrameWrapper is pool managed or not?

Reply:
1. Unfortunately, we can't directly check or log how many items are in a sync.Pool in Go. It's built to be a lightweight tool for handling temporary objects, so it doesn't keep track or expose how many items are in there.
2. As long as the stream settings don't change, we don't expect `avFrameWrapper`s to be garbage collected often i.e. the number of Get() is same/similar to the number of Put(). If there is a sudden decrease in the rate of Put() operations compared to Get(), it means that items are being retrieved from the pool faster than they are being returned, which can lead to a buildup of unused items in the pool. GC would then kick in. This could happen when we switch from a high frame rate to a lower one or a general slowdown in calling of `decode`/`onPacketRTP`.
3. I think 3 was answered [here](https://github.com/erh/viamrtsp/pull/41#discussion_r1720124104). 
4. I think all avFrameWrappers are eventually all pool managed. They are Put() into the pool through the `decode` -> `poolCleanupCallback` path. They are either freed via the finalizer garbage collection, or when stream settings changed and we manually free them with `d.dst.freeAVFrameWrapper()`.